lib/electron-run.js

```diff
@@ -15,25 +15,22 @@ try {
 var url = process.argv[2]
 
 electron.app.on('ready', function () {
+  var preload = path.join(__dirname, 'electron-inject.js')
   var options = {
     show: false,
     webPreferences: {
-      preload: path.join(__dirname, 'electron-inject.js')
-    }
+      preload: preload
+    },
+    preload: preload
   }
-  options.preload = options.webPreferences.preload
-  var mainWindow = new electron.BrowserWindow(options);
-
-  (mainWindow.loadURL || mainWindow.loadUrl).call(mainWindow, url)
-
-  electron.ipcMain.once('started', function () {
-    mainWindow.send('started')
-  })
+  var mainWindow = new electron.BrowserWindow(options)
 
   electron.ipcMain.on('zuulmessage', function (e, msg) {
     console.log(JSON.stringify(msg))
     if (msg.type === 'done') {
       electron.app.quit()
     }
-  })
+  });
```

Comment:
Make sure to register an event listener on `'zuulmessage'` _before_ doing `window.loadURL()`. Since the code in `electron-inject.js` doesn't run until the page has been loaded and since we have an event listener registered before, we should be fine.

Reply:
@vweevers WDYT about simplifying the require logic for `electron`? I guess the try-catch thing is there to support an older electron api?

Reply:
Probably. Current Electron docs don't mention `require('ipc')`, so let's remove it 🔥

Reply:
I'll do that in another PR.